test/odbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,361 @@
+  int length;
+  int precision;
+  int scale;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure column attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_LENGTH, // Get the length of the column (size of char in columns)
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, LENGTH_EXPECTED);
+  
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_PRECISION, // Get the precision of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &precision); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(precision, PRECISION_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_SCALE, // Get the scale of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &scale); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(scale, SCALE_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_TYPE_NAME, // Get the type name of the column
+                          name,
+                          BUFFER_SIZE,
+                          NULL,
+                          NULL); 
+  ASSERT_EQ(string(name), NAME_EXPECTED);
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+}
+
+TEST_F(PSQL_DataTypes_Bigint, Insertion_Success) {
+
+  const int BYTES_EXPECTED = 8;
+
+  long long int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> valid_inserted_values = {
+    "-9223372036854775808",
+    "9223372036854775807",
+    "3",
+    "NULL"
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+    {1, SQL_C_SBIGINT, &pk, 0, &pk_len},
+    {2, SQL_C_SBIGINT, &data, 0,  &data_len}
+  };
+
+  string insert_string{}; 
+  string comma{};
+  
+  for (int i = 0; i< valid_inserted_values.size(); ++i) {
+    insert_string += comma + "(" + std::to_string(i) + "," + valid_inserted_values[i] + ")";
+    comma = ",";
+  }
+
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
```

Comment:
Probably it would be good to add the 'Drop table if exists' statement before this

Same thing in other tests.

Reply:
If we do this, then I think it might be a good idea to just have the SetUp() run a drop if statement as well since the SetUp() runs before every tests.